salt/modules/virt.py

```diff
@@ -1967,6 +1967,42 @@ def update(name,
                 devices_node.extend(changes[dev_type]['sorted'])
                 need_update = True
 
+    # Update the serial and the console definitions
+    if 'serial_type' in kwargs:
+        serial_node = devices_node.find('serial')
+        if serial_node and serial_node.attrib['type'] != kwargs['serial_type']:
+            serial_node.set('type', kwargs['serial_type'])
+            need_update = True
+        if serial_node is None:
+            devices_node.append('serial').set('type', kwargs['serial_type'])
+            need_update = True
+
+        if 'console' in kwargs and kwargs['console'] == False:
+            if devices_node.find('console'):
+                devices_node.remove('console')
+                need_update = True
+        else:
+            console_node = devices_node.find('console')
+            if console_node == None:
+                console_node = devices_node.append('console')
+
+            if console_node.attrib('type') != kwargs['serial_type']:
+                console_node.set('type', kwargs['serial_type'])
+                need_update = True
+
+            source_node = console_node.find('source')
+            if kwargs['serial_type'] == 'tcp':
+                telnet = kwargs['telnet_port'] if 'telnet_port' in kwargs else 23023 # FIXME: use random unused port
```

Comment:
I know that code comes from `virt.init()` but I'm honestly not at ease with a hard-coded port number. Even picking a random unused port seems rather fragile. What about simply ignoring devices where that value is missing and logging an error?